src/qibolab/instruments/qblox/acquisition.py

```diff
@@ -3,24 +3,9 @@
 
 import numpy as np
 
-SAMPLING_RATE = 1
-
+from ...pulses.shape import demodulate
```

Comment:
```suggestion
from qibolab.pulses.shape import demodulate
```

Reply:
Do you have a reason to uses global imports?

It seems like there was a reason at the time of Python 2, but since Python 3 relative imports are perfectly fine, and I'd be in favor of thorough adoption for a few benefits:
- it makes clear it's an internal import
- frequently it avoids long import paths for things that are nested in the same place
- conversely, it makes explicit when the import is non-local
- it is convenient for moving subpackages around (since references are relative)

Even NumPy moved to relative imports...

Reply:
No fundamental reason, other than doing things consistently throughout codebase. Although, after adding the suggestion here I noticed that the codebase is not consistent anyways. These type of things can be enforced via formatters and style checkers, but this is a really minor thing. Don't think it needs a lengthy discussion.

Reply:
> These type of things can be enforced via formatters and style checkers, but this is a really minor thing.

Agreed.

> Don't think it needs a lengthy discussion.

That's why I'm limiting it to my code. At least for the time being.

Reply:
I would actually prefer that relative imports be used only for files in the same directory.

Reply:
> I would actually prefer that relative imports be used only for files in the same directory.

I see your point, but it misses part of the convenience.

It's not absurd: sometimes the mutual relation is limited to being in the same package (as in this case), so it makes more sense to start from top-level rather than bottom.
However, at this point I just like to remark the distinction from internal vs external import, and I can avoid discussing it case by case.

But if we wanted a criterion, I'd rather choose "they are in the same *top-level* subpackage/directory". E.g. if they are both in `instruments`, I'd use the relative (often you mostly navigate up in the hierarchy).

Reply:
> I would actually prefer that relative imports be used only for files in the same directory.

Btw, I know that writing is sometimes annoying, but could you try to make an effort to provide your motivations, together with your personal preferences?

Reply:
I think you already described all possible scenarios, and I didn't even think this far when adding the suggestion :D. It is not exact science, and boundaries get blurry as the directory hierarchy grows. I believe we are better off with simple intuition. Now that I think about it, I guess, a rule of thumb that I would use would be trying to answer this question "if the directory hierarchy were to change, is there a reason that would motivate to change it so that the relative locations of the two things under discussion remains the same?". If the answer is no, then I would use a global import. For this case I think the answer is no.

Reply:
> "if the directory hierarchy were to change, is there a reason that would motivate to change it so that the relative locations of the two things under discussion remains the same?"

Agreed, that was the idea behind my *top-level*. I.e. I'd like to assume I can always move/rename a subpackage (including top-level ones) without changing anything inside.

Here I pushed it too far, possibly, because I'd like to be able to *rename* `qibolab` itself (in Rust you could use the keyword `crate` to name top-level, without naming it).
But I agree this is not likely to happen at all, and it's rather impractical.